src/DataObject/DataObjectDocument.php

```diff
@@ -123,6 +123,14 @@ public function shouldIndex(): bool
     {
         $dataObject = $this->getDataObject();
 
+        // Allow DataObjects to completely override the indexing decision if necessary
+        if ($dataObject->hasMethod('shouldIndex')) {
```

Comment:
I'm tempted to call this `overrideShouldIndex()` (or something to that effect), otherwise I'm not sure that the implication is immediately clear, especially when we already have `canIndexInSearch()` which is supplementary to `shouldIndex()` (rather than an override).

Reply:
Yeah I did think about the naming of this, and my original conclusion was that because this completely replaces `shouldIndex`'s functionality, that it was effectively like a child class replacing a parent's method, and should be named as such - but will happily change it for a bit more clarity. (I'll go with `overrideShouldIndex()`, but I wonder if something like `mustIndex()` would reasonably articulate the difference instead?)

Reply:
`mustIndex()` sounds to me like it forces a return of `true`, rather than something that is conditionally true.

@blueo is back at work today, so curious what you think 😄 

#namingthingsishard

Reply:
Oh heck yea, an `interface` sounds like a great solution for this. Love that idea.

... What do we call the `interface`? 😂

```php
<?php

namespace SilverStripe\Forager\Interfaces;

interface SuperCoolName
{
    
    public function shouldIndex(): bool;

}
```

`DataObjectDocument` changes to something like:
```php
public function shouldIndex(): bool
{
    $dataObject = $this->getDataObject();

    if (in_array(SuperCoolName::class, class_implements($dataObject))) {
        return $dataObject->shouldIndex();
    }

    ... original stuff
}
```

Reply:
I'm thinking something like `Indexable` which leaves it a bit open for other things to be added fits with `Configurable` `Injectable` etc.

Reply:
I've gone with `DataObjectSelfDeterminesIndexability`, which is a bit of a mouthful, but I think clearly describes the purpose. We could alternatively try to align a bit better with existing interface names, e.g. `DataObjectIndexabilityHandler`... thoughts?

Reply:
yeah it it does what it says on the tin - just a big tin ;) `ShouldIndexProvider`, `ShouldIndexHandler`, `DataObjectIndexable`, `IndexableDelegate` ? Sort of reads like `this class looks after the ability to index`? 

 This probably isn't just for data objects - though that's our only use case rn (I can see it being split to Versioned/unversioned data objects at some point too).

Reply:
Ah, missed your comment sorry @blueo - `Indexable` feels a bit too generic though? 🤔

Reply:
Have settled on `IndexableHandler` - feels a bit generic, but the docblock articulates its (current) usage so I think it'll do the trick.

Reply:
@blueo @Cheddam 

Adding my $0.02 for the record, but you guys are already on it 😂 

I'm keen for this interface to be about just this one thing. I don't really see us adding new methods into that interface - because that would become a breaking change (similar scenario we had with Discoverer v1 as soon as we wanted to add support for spelling suggestions).

Also keen to leave out any mention of `DataObject` (even though that's all we can index atm).

With those 2 points in mind, I think I'm digging `IndexableHandler` and `IndexableDelegate`. I actually really like "delegate", but I wonder if "handler" might be a more commonly used, English-is-maybe-not-your-first-language word to describe it.

So... `IndexableHandler` wins?